core/lib/dr_tools.h

```diff
@@ -48,6 +48,12 @@ DR_API
  * \warning This context cannot be used as the drcontext for a thread
  * running under DR control!  It is only for standalone programs that
  * wish to use DR as a library of disassembly, etc. routines.
+ * \warning This context is not fully thread-safe as it stores
+ * the current ISA mode DR uses in a global variable.
+ * Hence, having different threads (e.g., in a parallel analysis tool)
+ * setting different ISA modes at the same time results in a race condition.
+ * Currently we require parallel analysis tools to use their own lock/unlock
+ * mechanism to avoid such race condition when setting the ISA mode.
```

Comment:
We should also clarify that the GLOBAL_DCONTEXT uses DR global heap which has its own lock, so the user doesn't need to use locks around decoder calls that may allocate/free memory.

But we start getting into various implementation details:
* that dr_standalone_init returns GLOBAL_DCONTEXT (which also isn't true for -DSTANDALONE_UNIT_TEST)
* which decoder APIs have heap alloc/free calls, vs accesses to other race-prone data like isa_mode. How would a user decide which decoder API calls need explicit locks?

Reply:
@derekbruening Thoughts on how we can include these clarifications in the API documentation?

Reply:
I don't think we need to worry about our internal unit tests in these docs.

Given that AArch32 seems to be problematic for basic decoding and encoding maybe we should just fix this and fix #6002 at the same time by requiring an explicit init call in every thread (having it create a new dcontext) and dropping the implicit init support?  There may be complexities with a new dcontext in standalone mode: will some components try to do managed-mode-only things with it at init time?  Another challenge of not having clear separation between managed mode and regular library use.

Reply:
Updated the doc about AArch32 encoding/decoding for now.